src/ert/_c_wrappers/job_queue/workflow_job.py

```diff
@@ -80,27 +126,63 @@ def _make_arg_types_list(
             return []
 
     @classmethod
-    def fromFile(cls, config_file, name=None):
-        if os.path.isfile(config_file) and os.access(config_file, os.R_OK):
-            if name is None:
-                name = os.path.basename(config_file)
+    def _file_to_workflow_args(cls, config_file, name=None, use_new_parser=True):
+        if not name:
+            name = os.path.basename(config_file)
+
+        if use_new_parser:
+            new_content_dict = new_workflow_job_parser(config_file)
 
-            content = _config_parser.parse(config_file)
+            new_types_list = cls._make_arg_types_list_new(new_content_dict)
+            return (
+                name,
+                new_content_dict.get("INTERNAL"),
+                new_content_dict.get("MIN_ARG"),
+                new_content_dict.get("MAX_ARG"),
+                new_types_list,
+                new_content_dict.get("EXECUTABLE"),
+                str(new_content_dict.get("SCRIPT"))
+                if "SCRIPT" in new_content_dict
+                else None,
+            )
+        else:
+            old_content = _config_parser.parse(config_file)
 
             def optional_get(key):
-                return content.getValue(key) if content.hasKey(key) else None
+                return old_content.getValue(key) if old_content.hasKey(key) else None
 
             max_arg = optional_get("MAX_ARG")
 
-            return cls(
+            old_arg_types_list = cls._make_arg_types_list(old_content, max_arg)
+            return (
                 name,
                 optional_get("INTERNAL"),
                 optional_get("MIN_ARG"),
                 max_arg,
-                cls._make_arg_types_list(content, max_arg),
+                old_arg_types_list,
                 optional_get("EXECUTABLE"),
                 optional_get("SCRIPT"),
             )
+
+    @classmethod
+    def fromFile(cls, config_file, name=None):
+        if os.path.isfile(config_file) and os.access(config_file, os.R_OK):
+            new = cls._file_to_workflow_args(config_file, name, use_new_parser=True)
+            old = cls._file_to_workflow_args(config_file, name, use_new_parser=False)
+
+            if new[4] != old[4]:
+                # typelists differ in a few cases
+                # as the new parser infers one from
+                # set min args / max args, but old
+                # parser leaves it as blank
+                # Test case: MIN_ARG = 1 and unspecified MAX_ARG
+                # and ARG_TYPE
+                pass
+            elif new != old:
```

Comment:
We don't want to raise an exception if they are unequal. We just want to log the difference. Also, any exception that happens in the comparison should not be noticed by the user. See https://github.com/equinor/ert/blob/0f4b6059c58c685a0c5bf27e545109bddfa943fd/src/ert/gui/main.py#L80

Reply:
Done, now only logging the difference between them